Modules/aiida_ensemble.py

```diff
@@ -0,0 +1,125 @@
+        method.
+
+        Parameters
+        ---------
+            pw_code:
+                The string associated with the AiiDA code for `pw.x`
+            profile: 
+                AiiDA profile to use during the calculations. If None, the default is used
+            protocol:
+                The protocol to be used; available protocols are 'fast', 'moderate' and 'precise'
+            options:
+                The options for the calculations, such as the resources, wall-time, etc.
+            overrides:
+                The overrides for the get_builder_from_protocol
+            kwargs:
+                The kwargs for the get_builder_from_protocol
+        """
+        from aiida import load_profile
+        from aiida.engine import submit
+        from aiida.plugins import WorkflowFactory
+        from aiida.orm import StructureData
+        
+        PwBaseWorkChain = WorkflowFactory('quantumespresso.pw')
+    
+        if profile:
+            load_profile(profile)
+        else:
+            load_profile()
+        
+        # Check if not all the calculation needs to be done
+        n_calcs = np.sum( self.force_computed.astype(int)) 
+        computing_ensemble = self
+
+        if overrides:
+            try:
+                tstress = overrides['pw']['parameters']['CONTROL']['tstress']
+                self.has_stress = tstress
+            except KeyError:
+                self.has_stress = True # by default we calculate stresses with the `get_builder_from_protocol`
+
+        # Check wheter compute the whole ensemble, or just a small part
+        should_i_merge = False
+        if n_calcs != self.N:
+            should_i_merge = True
+            computing_ensemble = self.get_noncomputed()
+            self.remove_noncomputed()
+            
+        # ============= HERE AIIDA PART ============= #
+        structures_data = [StructureData(ase=cc.get_ase_atoms()) for cc in self.structures]
+        workchains = []
+        
+        for i, structure in enumerate(structures_data):
+            builder = PwBaseWorkChain.get_builder_from_protocol(
+                code=pw_code,
+                structure=structure,
+                options=options,
+                overrides=overrides,
+                **kwargs
+            )
+            builder.label = str(i)
+            workchains.append(submit(builder))
+        
+        workchains_success = []
+        workchains_failed = []
+
+        while(workchains):
+            for workchain in workchains:
+                if workchain.is_finished:
+                    if workchain.is_failed:
+                        workchains_failed.append(workchain)
+                    else:
+                        workchains_success.append(workchain)
+                    workchains.remove(workchain) # here it may be critical
+
+            time.sleep(60) # wait before checking again
+        
+        # ---- TO DO
```

Comment:
@mesonepigreco, could you help me on how to store the quantities here?

I saw there is a `all_properties`, so should I fill that as well? What it is in particular?

Moreover, what happens if not all the structures converged correctly, e.g. cause they are very tricky to converge? Does the entire SSCHA engine stops, or it can continue with fewer structures?

Reply:
A jupyter is nice because you can  add a bit of explanation in markdown text

Reply:
For the dependency, add them in the github workflow:

.git/workflows/testsuite-something.yml

Add the command to install the extra dependencies in the build so that the testsuite passes.